includes/class-newspack-listings-products.php

```diff
@@ -231,14 +351,1426 @@ public static function get_products() {
+	 * @return string Cleaned URL.
+	 */
+	public static function get_base_url() {
+		return isset( $_SERVER['REQUEST_URI'] ) ? site_url( strtok( sanitize_text_field( $_SERVER['REQUEST_URI'] ), '?' ) ) : get_permalink( get_option( 'woocommerce_myaccount_page_id' ) );
+	}
+
+	/**
+	 * Active premium subscriptions grant customers the ability to create up to 10 Marketplace or Event listings for free.
+	 * Show controls to create and manage these listings in the Subscription account page.
+	 *
+	 * @param WC_Subscription $subscription Subscription object for the subscription whose status has changed.
+	 */
+	public static function listing_subscription_manage_premium_listings( $subscription ) {
+		$subscription_id  = $subscription->get_id();
+		$is_active        = 'active' === $subscription->get_status();
+		$is_premium       = $is_active && get_post_meta( $subscription_id, self::SUBSCRIPTION_META_KEYS['is_premium'], true );
+		$customer_id      = $subscription->get_user_id();
+		$premium_listings = get_posts(
+			[
+				'meta_key'       => self::POST_META_KEYS['listing_subscription'],
+				'meta_value'     => $subscription->get_id(), // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_value
+				'post_status'    => [ 'draft', 'future', 'pending', 'private', 'publish' ],
+				'post_type'      => [
+					Core::NEWSPACK_LISTINGS_POST_TYPES['event'],
+					Core::NEWSPACK_LISTINGS_POST_TYPES['marketplace'],
+				],
+				'posts_per_page' => self::TOTAL_FREE_LISTINGS, // phpcs:ignore WordPress.WP.PostsPerPage.posts_per_page_posts_per_page
+			]
+		);
+
+		// If not a premium subscription and there are no previously created premium listings, no need to show the UI.
+		if ( ! $is_premium && 0 === count( $premium_listings ) ) {
+			return;
+		}
+
+		$remaining       = self::TOTAL_FREE_LISTINGS - count( $premium_listings );
+		$base_url        = self::get_base_url();
+		$marketplace_url = wp_nonce_url(
+			add_query_arg(
+				[
+					'customer_id'     => $customer_id,
+					'subscription_id' => $subscription_id,
+					'create'          => 'marketplace',
+				],
+				$base_url
+			),
+			self::CREATE_LISTING_NONCE
+		);
+		$event_url       = wp_nonce_url(
+			add_query_arg(
+				[
+					'customer_id'     => $customer_id,
+					'subscription_id' => $subscription_id,
+					'create'          => 'event',
+				],
+				$base_url
+			),
+			self::CREATE_LISTING_NONCE
+		);
+
+		?>
+			<h2><?php echo esc_html( __( 'Premium subscription listings', 'newspack-listings' ) ); ?></h2>
+			<p>
+				<?php
+					echo esc_html(
+						sprintf(
+							// translators: explanation of premium subscription benefits.
+							__( 'A premium subscription lets you create up to %1$d free Marketplace or Event listings. %2$s', 'newspack-listings' ),
+							self::TOTAL_FREE_LISTINGS,
+							$remaining && $is_premium ?
+								// translators: explanation of remaining free listings.
+								sprintf( __( 'You have %d free listings remaining.', 'newspack-listings' ), $remaining ) :
```

Comment:
Using the word "free" here feels weird, since the buyer just paid for this. These should rather be "listings in your subscription", or something along these lines.

Reply:
Good point—8eb66b8 updates this to say "included" and "additional" instead of "free."